app/src/main/java/com/autoscout/moviedb/moviedetails/MovieDetailsFragment.kt

```diff
@@ -0,0 +1,145 @@
+        val voteAverage = arguments?.get("voteAverage").toString()
+        val voteCount = arguments?.get("voteCount").toString()
+        val backdropPath = arguments?.get("backdropPath")
+
+        val toolbar = view.findViewById(R.id.toolbar) as Toolbar
+        toolbar.title = movieName
+        (activity as AppCompatActivity).setSupportActionBar(toolbar)
+        (activity as AppCompatActivity).supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        val posterImg = view.findViewById<ImageView>(R.id.movie_poster)
+        Glide
+                .with(view.context)
+                .load(BASE_URL_IMG + backdropPath)
+                .transition(DrawableTransitionOptions().crossFade())
+                .apply(RequestOptions().diskCacheStrategy(DiskCacheStrategy.ALL).centerCrop())
+                .into(posterImg)
+
+        (view.findViewById(R.id.title) as TextView).text = movieName
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            (view.findViewById(R.id.movie_release_date) as TextView).text = LocalDate.parse(movieReleaseDate).format(DateTimeFormatter.ofPattern("dd MMMM yyyy"))
+        }
+        (view.findViewById(R.id.vote_average) as TextView).text = voteAverage
+        (view.findViewById(R.id.vote_count) as TextView).text = voteCount
+
+        val favButton = view.findViewById<FloatingActionButton>(R.id.fav_icon)
+        favButton.setOnClickListener {
+            Toast.makeText(activity, it.context.getText(R.string.fav_added),
+                    Toast.LENGTH_LONG).show()
+        }
+
+        val castsRecyclerView: androidx.recyclerview.widget.RecyclerView = view.findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.recycler_view_cast_movie_detail).apply {
+            layoutManager = androidx.recyclerview.widget.LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+            adapter = movieCastsAdapter
+        }
+
+        val movieDetailsTextView: TextView = view.findViewById(R.id.movie_details_overview)
+        val showMoreButton: Button = view.findViewById(R.id.show_more)
+        showMoreButton.setOnClickListener(object : View.OnClickListener {
+
+            override fun onClick(v: View?) {
+                if (overviewExpanded) {
+                    val animation: ObjectAnimator = ObjectAnimator.ofInt(movieDetailsTextView, "maxLines", 3);
+                    animation.setDuration(200).start();
+                    showMoreButton.setText("Show more...")
+                    overviewExpanded = false
+                } else {
+                    val animation: ObjectAnimator = ObjectAnimator.ofInt(movieDetailsTextView, "maxLines", 40)
+                    animation.setDuration(200).start()
+                    showMoreButton.setText("Show less...")
+                    overviewExpanded = true
+                }
+            }
+        })
+
+
+        movieDetailsTextView.text = movieOverview
+
+        loadCasts()
+    }
+
+    private fun loadCasts() {
```

Comment:
With the click listeners in the adapter classes, I think you have learned a lot about Kotlin functions. Why don't you create a `MovieDetailApiCallManager` class? Then just by passing in the handleSuccess() and handleError() methods you can have a cleaner and readable code in Fragment classes

Reply:
You can do a similar thing in MovieListFragment as well. Don't forget that we will also use DI in the future so better to have separate classes for each task.